src/proxy-ui-api/frontend/src/views/KeysAndCertificates/SignAndAuthKeys/KeyRow.vue

```diff
@@ -65,29 +63,41 @@ export default Vue.extend({
       type: Object as Prop<Key>,
       required: true,
     },
-    hasPermission: {
-      type: Boolean,
-    },
     tokenLoggedIn: {
       type: Boolean,
     },
   },
-  methods: {
-    disableGenerateCsr(key: Key): boolean {
-      if (!this.tokenLoggedIn) {
-        return true;
+  computed: {
+    showGenerateCsr(): boolean {
+      // Check if the user has permission to see generate csr action
+      if (this.tokenKey.usage === 'AUTHENTICATION') {
+        return this.$store.getters.hasPermission(
+          Permissions.GENERATE_AUTH_CERT_REQ,
+        );
       }
+      // If key usage is not auth then it has to be sign
+      return this.$store.getters.hasPermission(
```

Comment:
Does this work OK if key does not yet have a defined usage (no CSRs have been created) and user has only `generate_auth_cert_req` permission, not `generate_sign_cert_req`? (there is no such role at the moment, though, which would hide the issue)

Reply:
If the key doesn't have a usage type it ends up in UnknownKeysTable.vue. That doesn't use the KeyRow component.

Reply:
ah, gotcha :+1: